src/pages/RFOX/components/AddressSelection.tsx

```diff
@@ -177,18 +191,39 @@ export const AddressSelection: FC<AddressSelectionProps> = ({
   }, [accountIdsByAccountNumberAndChainId, stakingAssetAccountNumber])
 
   const maybeRuneAccountId = useMemo(() => {
-    if (selectedAddress) return toAccountId({ account: selectedAddress, chainId: thorchainChainId })
+    if (selectedAddress && !shouldDisableAccountDropdown)
```

Comment:
It was the reason of a bug where displayed address was wrong if you were switching accounts